pkg/controller/policyrecommendation/policyrecommendation_controller.go

```diff
@@ -0,0 +1,429 @@
+		return reconcile.Result{RequeueAfter: 10 * time.Second}, nil
+	}
+
+	// Query for the installation object.
+	variant, network, err := utils.GetInstallation(ctx, r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.SetDegraded(operatorv1.ResourceNotFound, "Installation not found", err, reqLogger)
+			return reconcile.Result{}, err
+		}
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error querying installation", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	pullSecrets, err := utils.GetNetworkingPullSecrets(network, r.client)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Failed to retrieve pull secrets", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	esClusterConfig, err := utils.GetElasticsearchClusterConfig(ctx, r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.SetDegraded(operatorv1.ResourceNotReady, "Elasticsearch cluster configuration is not available, waiting for it to become available", err, reqLogger)
+			return reconcile.Result{}, nil
+		}
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Failed to get the elasticsearch cluster configuration", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	managementCluster, err := utils.GetManagementCluster(ctx, r.client)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error reading ManagementCluster", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	managementClusterConnection, err := utils.GetManagementClusterConnection(ctx, r.client)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Error reading ManagementClusterConnection", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	isManagedCluster := managementClusterConnection != nil
+
+	if managementClusterConnection != nil && managementCluster != nil {
+		err = fmt.Errorf("having both a ManagementCluster and a ManagementClusterConnection is not supported")
+		r.status.SetDegraded(operatorv1.ResourceValidationError, "", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	secretsToWatch := []string{
+		render.ElasticsearchPolicyRecommendationUserSecret,
+	}
+	esSecrets, err := utils.ElasticsearchSecrets(ctx, secretsToWatch, r.client)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			r.status.SetDegraded(operatorv1.ResourceNotReady, "Elasticsearch secrets are not available yet, waiting until they become available", err, reqLogger)
+			return reconcile.Result{}, nil
+		}
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Failed to get Elasticsearch credentials", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	certificateManager, err := certificatemanager.Create(r.client, network, r.clusterDomain)
+	if err != nil {
+		r.status.SetDegraded(operatorv1.ResourceCreateError, "Unable to create the Tigera CA", err, reqLogger)
+		return reconcile.Result{}, err
+	}
+
+	var managerInternalTLSSecret certificatemanagement.CertificateInterface
+	if !isManagedCluster {
```

Comment:
I think this should be if `managementCluster != nil`
I am not sure that secret `render.ManagerInternalTLSSecretName` exists for standalone clusters.